app/lib/azul_search_service.rb

```diff
@@ -229,29 +229,39 @@ def self.is_analysis_file(file_summary)
   def self.get_file_summary_info(accessions)
     client = ApplicationController.hca_azul_client
     file_query = { 'project' => { 'is' => accessions } }
-    results = client.projects(query: file_query)
-    results['hits'].map do |entry|
-      entry_hash = entry.with_indifferent_access
-      project_hash = entry_hash[:projects].first # there will only ever be one project here
-      accession = project_hash[:projectShortname]
-      result = {
-        study_source: 'HCA',
-        name: project_hash[:projectTitle],
-        accession: accession,
-        description: project_hash[:projectDescription],
-        studyFiles: [
-          {
-            project_id: project_hash[:projectId],
-            file_type: 'Project Manifest',
-            count: 1,
-            upload_file_size: 1.megabyte, # placeholder filesize as we don't know until manifest is downloaded
-            name: "#{accession}.tsv"
-          }
-        ]
-      }.with_indifferent_access
-      project_file_info = extract_file_information(entry_hash)
-      result[:studyFiles] += project_file_info if project_file_info.any?
-      result
+    if client.query_too_large?(file_query)
+      # cut query in half to prevent HTTP 413
+      queries = accessions.each_slice((accessions.size / 2.0).round).map { |list| { 'project' => { 'is' => list } } }
+    else
+      queries = [{ 'project' => { 'is' => accessions } }]
+    end
+    all_results = []
+    queries.each do |query|
```

Comment:
would this be a good case for Parallel.map so that we can fetch the twin results in parallel?  Or would the performance impact be negligible?

Reply:
Interesting question!  I did a little experimentation locally, and here's what I found:

```
Benchmark.bm do |x|
  x.report { AzulSearchService.get_file_summary_info(projects)  }
end

SERIAL
       user     system      total        real
   0.498807   0.014046   0.512853 (  3.696607)

PARALLEL
       user     system      total        real
   0.503768   0.017950   0.521718 (  2.022230)
```

It's interesting that the numbers all look the same, except that the "real" time is close to double.  I ran this several times, and the results hold.  I think getting an extra 1.5s is worth it.  Great idea!

Reply:
[f2f0973](https://github.com/broadinstitute/single_cell_portal_core/pull/1521/commits/f2f0973de9b76bb6eaeba298b587fbe69306ba57)